src/main/java/nextstep/web/repository/ReservationDao.java

```diff
@@ -0,0 +1,69 @@
+package nextstep.web.repository;
+
+import lombok.RequiredArgsConstructor;
+import nextstep.domain.Reservation;
+import nextstep.domain.Theme;
+import nextstep.web.exception.BusinessException;
+import nextstep.web.exception.CommonErrorCode;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Repository
+@RequiredArgsConstructor
+@Transactional
+public class ReservationDao implements ReservationRepository {
+
+    public static final String TABLE_NAME = "reservation";
+    public static final String KEY_COLUMN_NAME = "id";
+    private final JdbcTemplate jdbcTemplate;
+
+    private final RowMapper<Reservation> actorRowMapper = (resultSet, rowNum) -> Reservation.from(resultSet);
+
+    public Long save(Reservation reservation) {
+        SimpleJdbcInsert simpleJdbcInsert = new SimpleJdbcInsert(jdbcTemplate)
+                .withTableName(TABLE_NAME)
+                .usingGeneratedKeyColumns(KEY_COLUMN_NAME);
+
+        Number key = simpleJdbcInsert.executeAndReturnKey(prepareParams(reservation));
+
+        return key.longValue();
+    }
+
+
+    public Reservation findById(Long id) {
+        String sql = "SELECT * FROM reservation WHERE ID = ?;";
+        List<Reservation> reservations = jdbcTemplate.query(sql, actorRowMapper, id);
+        if (reservations.isEmpty()) {
+            throw new BusinessException(CommonErrorCode.RESOURCE_NOT_FOUND);
+        }
+
+        return reservations.get(0);
+    }
+
+    public void deleteById(Long id) {
+        String sql = "DELETE FROM reservation WHERE ID = ?;";
+        if (jdbcTemplate.update(sql, id) == 0) {
+            throw new BusinessException(CommonErrorCode.RESOURCE_NOT_FOUND);
+        }
+    }
+
+    private static Map<String, Object> prepareParams(Reservation reservation) {
+        Theme theme = reservation.getTheme();
+        Map<String, Object> params = new HashMap<>();
+
+        params.put("date", reservation.getDate());
+        params.put("time", reservation.getTime());
+        params.put("name", reservation.getName());
+        params.put("theme_name", theme.getName());
+        params.put("theme_desc", theme.getDesc());
+        params.put("theme_price", theme.getPrice());
+        return params;
```

Comment:
Java 11 이라면 Map.of, Map.ofEntries 를 활용해도 괜찮을 것 같아요~
```suggestion
        return Map.of(
          "date", reservation.getDate(),
          "time", reservation.getDate(),
          // ~~~
          "theme_price", theme.getPrice()
        );
```

보기에도 간결해질 수 있고, ImmutableCollection 을 내부적으로 반환하기 때문에 의도를 더 명확히 나타낼 수도 있을 것 같아요!

참고로 java8 을 사용하고 있는 저희팀은 guava 등 라이브러리 도움을 받을때도 있습니다~

Reply:
```java
    private static Map<String, Object> prepareParams(Reservation reservation) {
        Theme theme = reservation.getTheme();

        return Map.of(
                "date", reservation.getDate(),
                "time", reservation.getTime(),
                "name", reservation.getName(),
                "theme_name", theme.getName(),
                "theme_desc", theme.getDesc(),
                "theme_price", theme.getPrice()
        );
    }
```
너무 깔끔해졌습니다ㅠㅠ. List.of는 11에서부터 사용하고 있었는데, Map.of는 왜 생각도 못했을까요...
감사합니다!